app/src/main/java/org/oppia/app/topic/play/StorySummaryAdapter.kt

```diff
@@ -109,27 +113,81 @@ class StorySummaryAdapter(
 
       val chapterList = storySummaryViewModel.storySummary.chapterList
       binding.chapterRecyclerView.adapter = ChapterSummaryAdapter(chapterList, chapterSummarySelector)
+      if (currentExpandedChapterListIndex != null && currentExpandedChapterListIndex == position) {
+        val rotate = AnimationUtils.loadAnimation(context, R.anim.rotation_clockwise_180)
+        binding.chapterListDropDownIcon.startAnimation(rotate)
+        expand(binding.chapterListContainer)
+      } else {
+        val rotate = AnimationUtils.loadAnimation(context, R.anim.rotation_anti_clockwise_180)
+        binding.chapterListDropDownIcon.startAnimation(rotate)
+        collapse(binding.chapterListContainer)
+      }
 
       binding.root.setOnClickListener {
-        val previousIndex: Int? = currentExpandedChapterListIndex
-        currentExpandedChapterListIndex =
-          if (currentExpandedChapterListIndex != null && currentExpandedChapterListIndex == position) {
-            null
-          } else {
-            position
-          }
-        expandedChapterListIndexListener.onExpandListIconClicked(currentExpandedChapterListIndex)
-        if (previousIndex != null && currentExpandedChapterListIndex != null && previousIndex == currentExpandedChapterListIndex) {
-          notifyItemChanged(currentExpandedChapterListIndex!!)
+        val previousItem = currentExpandedChapterListIndex
+        if (binding.chapterListContainer.isVisible) {
+          currentExpandedChapterListIndex = null
         } else {
-          if (previousIndex != null) {
-            notifyItemChanged(previousIndex)
+          currentExpandedChapterListIndex = position
+        }
+        expandedChapterListIndexListener.onExpandListIconClicked(currentExpandedChapterListIndex)
+        notifyItemChanged(position)
+        if (previousItem != null && previousItem != position) {
+          notifyItemChanged(previousItem)
+        }
+      }
+    }
+
+    private fun expand(chapterListContainer: View) {
```

Comment:
I'm a bit confused by this. Why can't we rely on RecyclerView's own item animator for this? These animations seem a bit complex, and should really be tested directly since if they regress it will be hard for us to detect. I'd rather we rely on existing animation systems rather than creating our own Animation classes, if possible.

Reply:
I am actually not able to understand how can we do this using RecyclerView's animator?
Also, the main point behind introducing these animations were two fold:
1. Make expand/collapse animation smooth.
2. The animation time of expand, and arrow rotate all should be in sync.

I don't have much experience of automated animation, can you just give information regarding RecyclerView's item animator in more detail?
Thanks.

Reply:
Sure. I was thinking specifically DefaultItemAnimator, but I'm guessing that might not do the right thing. In which case, maybe something like https://stackoverflow.com/a/37453839 or another animator library might make this a bit easier if it's possible.

Reply:
This animation is application on recyclerview item. But here we want to expand and contract a linear layout, which is again within a recyclerview item.